### ISSUE TITLE
`mixer bundle edit` can create a bundle with invalid name

### ISSUE THREAD

**Issue by phmccarty** [clearlinux/mixer-tools#448]
Steps to reproduce (from a empty workspace):

```
$ mixer init
$ mixer bundle edit foo.bar
<edit bundle definition and save>
Error parsing bundle "foo.bar": Invalid bundle name "foo.bar" derived from file "/home/pmccarty/mix/local-bundles/foo.bar"
Would you like to edit as-is, revert and edit, or skip [Edit/Revert/Skip]?:
```

No matter what option I select (Edit, Revert, or Skip), I cannot get mixer out of this bad state. For example, if I choose Skip, I can successfully remove the `foo.bar` bundle, but mixer then gets confused about the .orig file it left behind:

```
$ mixer bundle remove --local foo.bar
Warning: Using Format=25 from mixer.state for this build.
Removing bundle file for "foo.bar" from local-bundles
Bundle "foo.bar" not found in mix; skipping
$ mixer bundle list
ERROR: Invalid bundle name "foo.bar.orig" derived from file "/home/pmccarty/mix/local-bundles/foo.bar.orig"
```

Perhaps mixer should not try to edit a freshly created bundle definition if the bundle name is invalid?

**Comment by reaganlo** [comment]
@phmccarty @tmarcu Apologies, but this is going to be a long winded question.

Currently if a user creates a bundle using **Mixer** with an invalid name, the invalid bundle as well as its back up is created. I've done a quick fix where the invalid bundle as well as its backup does not get created.

But while testing a scenario where the user **manually** copies bundles from elsewhere with invalid names to the local-bundles folder, I came across another issue that "mixer bundle list" generates an invalid bundle name error. Ideally it should **not effect** this command because the invalid bundles are only in the local-bundles folder and not in the mix (mixbundles file). While working on a fix for this, I noticed that bundles are validated at various places in the code but not in a consistent manner. I can try to make it consistent which brings me to the main question - why do we need two levels of bundle validation (Basic and Strict). Can we just have one level (Strict) that is used consistently throughout the code.
